core/templates/pages/blog-admin-page/blog-admin-page.component.ts

```diff
@@ -119,7 +119,11 @@ export class BlogAdminPageComponent implements OnInit {
         formResponse.newRole);
       this.refreshFormData();
     }, errorResponse => {
-      this.statusMessage = 'Server error: ' + errorResponse;
+      if (errorResponse.includes('duplicate')) {
+        errorResponse = 'User ' + formResponse.username + ' is already' +
+        ' assigned as ' + formResponse.newRole;
+      }
```

Comment:
Why not just correct the backend error message?

Reply:
The backend error message is more generic, and might be well appropriate for some other cases, so I did not touch the backend message, rather modified the frontend message as per this situation.

Reply:
Considering that we are planning to surface the errors to the user, should we make the error messages less generic? The errors seem like validation errors. Perhaps see how validation errors in the editor page are handled (https://github.com/oppia/oppia/blob/develop/core/controllers/editor.py).

Reply:
I think the issue wants to make the error message more understandable for a general user, as the validation error message seems very technical for a general user. I followed your reference for handling validation error, using InvalidInputException, but the error message displayed is still the same. 
So should I change the message in the backend itself?

Reply:
Hm, looking into the code flow, I think we should update add_user_role and remove_user_role in user_services such that we explicitly check for these cases and report appropriate errors before saving the model. Currently, no explicit validation is done. All the validation happens at the model level in the last step when trying to update the model -- this is not ideal and probably explains why the errors are not very user friendly.

Reply:
The error in the issue needs adding this message in the add_user_role method, so I have updated it there. PTAL.